docs: Minify README language

mieubrisse opened the pull request:
Make the README starting section less wordy, and use our "distributed application development platform" category

Changelog picked up from commits here:

docs: Minify README language

galenmarchetti commented on CHANGELOG.md:
```diff
@@ -4,6 +4,8 @@
 * Refactored the big `for` loop inside `TestSuiteRunner.RunTests` into a separate helper function
 * Use `defer` to stop the testnet after it's created, so we stop it even in the event of unanticipated panics
+* Allow tests to stop network nodes
+* Force the user to provide a static configuration ID, so they can reference it later in their tests
```
can you add more details here? configuration ID for what? network, services, each test, etc

mieubrisse replied:
Done

galenmarchetti commented on commons/docker/docker_manager.go:
```diff
@@ -58,10 +59,6 @@ func (manager DockerManager) CreateNetwork(subnetMask string, gatewayIP string)
-//   ~ 2020-05-30
-/*
 func (manager DockerManager) CreateVolume(volumeName string) (pathOnHost string, err error) {
 	volumeConfig := volume.VolumeCreateBody{
```
oh baby

mieubrisse replied:
:D

galenmarchetti commented on commons/docker/docker_manager.go:
```diff
@@ -240,36 +237,24 @@ mapped to the host ports.
 Args:
 	usedPorts: a "set" of ports that the container will listen on (and which need to be mapped to host ports)
 	bindMounts: mapping of (host file) -> (mountpoint on container) that will be mounted at container startup
```
can we add a comment here that explains when we should be using bind mounts vs when we should be using volumes? right now it seems like duplicate functionality, although I imagine there's a reason we're retaining both

mieubrisse replied:
Yep! Added:
```
	bindMounts: Mapping of (host file) -> (mountpoint on container) that will be mounted at container startup (used when
		sharing data between the host filesystem - in our case, the test initializer - and a Docker container)
	volumeMounts: Mapping of (volume name) -> (mountpoint on container) that will be mounted at container startup (used
		when sharing data between containers). This is distinct from a bind mount because the host filesystem can't easily
		read from a Docker volume - you need to be inside a Docker container to do so.
```
LMK if that makes sense; biggest distinction is that as the host, you actually can't easily write to a Docker volume because Docker _itself_ lives inside a VM. So in the initializer when you create a volume, you get back a filepath that's relative to the Docker VM and doing File.Write will give you filepath not found.

galenmarchetti commented on commons/networks/service_network.go:
```diff
@@ -1,12 +1,155 @@
-type ServiceNetwork struct {
-	NetworkId string
+type ServiceNode struct {
+	IpAddr string
```
YESS

galenmarchetti commented on commons/networks/service_network.go:
```diff
@@ -1,12 +1,155 @@
+	return len(network.serviceNodes)
+}
+
+// Adds a service to the graph, with the specified dependencies (with the map used only as a set - the values are ignored)
```
IMO "dependencies" are not clear...can we add more information in this comment about what kinds of things constitute dependencies?

I know in the ava specific context its "serivices ids of bootstrappers", but in the generalized context it's not clear to me what I should be expect to be able to do with the dependencies

mieubrisse replied:
new docs for the function; lmk if this resolves:
```
Adds a service to the graph that depends on the services with the given IDs
Args:
	configurationId: The ID of the service configuration to use for creating the service
	serviceId: The ID to give the node in the network
	dependencies: A "set" of service IDs that the node-to-create depends on - i.e., whose information the node-to-create
		needs to start up. If the node-to-create doesn't depend on any other services, the dependencies map should be
		empty (not nil).
Return:
	An AvailabilityChecker for checking when the service is actually available
```

galenmarchetti commented on commons/testsuite/test_network_loader.go:
```diff
@@ -7,10 +7,19 @@ import (
+	Hook for the user to initialize the network to whatever state they'd like the Test to have
+	Args:
+		network: The network that the user should call AddService on
+	Returns: A map of serviceId -> availability checkers that indicate that the network is started if all checkers return
```
whoa nice

galenmarchetti commented on commons/networks/public_ip_tracker.go:
```diff
@@ -11,29 +11,36 @@ type FreeIpAddrTracker struct {
+Creates a new tracker that will dole out free IP addresses from the given subnet, making sure not to dole out any IPs
+from the list of already-taken IPs
+ */
+func NewFreeIpAddrTracker(subnetMask string, alreadyTakenIps []string) (ipAddrTracker *FreeIpAddrTracker, err error) {
```
why are we adding the "alreadyTakenIps" to this constructor? what would be the use of this parameter?

mieubrisse replied:
Yep so, this is used in the `testController`. The reason for it is that the TestController (who now does network construction) needs to know which IPs are already in use in the subnet that he lives inside, so he doesn't dole them out new services. As of right now, the two IPs that will be in use will be the Gateway IP and the test controller IP itself. This is also why GATEWAY_IP and TEST_CONTROLLER_IP are now arguments to the TestContorller

mieubrisse committed:
docs: Minify README language

Make the README starting section less wordy, and use our "distributed application development platform" category

mieubrisse committed:
Make README consistent with latest docs changes

mieubrisse committed:
Merge branch 'main' into mieubrisse-patch-1

mieubrisse committed:
Merge branch 'main' into mieubrisse-patch-1
